cmd/metal-api/internal/datastore/switch.go

```diff
@@ -196,3 +196,39 @@ func (rs *RethinkStore) ConnectMachineWithSwitches(m *metal.Machine) error {
 
 	return nil
 }
+
+// GetSwitchStatus get SwitchStatus for a given switch id
+func (rs *RethinkStore) GetSwitchStatus(id string) (*metal.SwitchStatus, error) {
+	rs.log.Infow("get state", "id", id)
+	var ss metal.SwitchStatus
+	err := rs.findEntityByID(rs.switchStatusTable(), &ss, id)
+	if err != nil && !metal.IsNotFound(err) {
+		return nil, err
+	}
+	if err != nil && metal.IsNotFound(err) {
+		return &metal.SwitchStatus{}, nil
+	}
```

Comment:
For consistency, I think this kind of error handling should go into the service layer. Other functions of this package, which do similar things, do not return empty responses on non-existent entities.

Reply:
done